tests/python/pants_test/backend/python/tasks/test_python_run_integration.py

```diff
@@ -44,11 +44,36 @@ def test_run_27_and_then_3(self):
       )
       self.assert_success(pants_run_3)
 
+  def test_run_3_by_option(self):
+    if self.skip_if_no_python('3'):
+      return
+
+    with temporary_dir() as interpreters_cache:
+      pants_ini_config = {'python-setup': {'interpreter_constraints': ["CPython>=2.7,<4"]}}
+      pants_run_3 = self.run_pants(
+        command=['run', '{}:echo_interpreter_version_3'.format(self.testproject),
+        '--python-setup-interpreter-constraints=CPython>=3'],
+        config=pants_ini_config
+      )
+      self.assert_success(pants_run_3)
+
+  def test_run_2_by_option(self):
+    if self.skip_if_no_python('3'):
+      return
+
+    with temporary_dir() as interpreters_cache:
+      pants_ini_config = {'python-setup': {'interpreter_constraints': ["CPython>=2.7,<4"]}}
+      pants_run_2 = self.run_pants(
+        command=['run', '{}:echo_interpreter_version_2'.format(self.testproject),
+        '--python-setup-interpreter-constraints=CPython<3'],
+        config=pants_ini_config
+      )
+      self.assert_success(pants_run_2)
+
   def test_die(self):
     command = ['run',
                '{}:die'.format(self.testproject),
-               '--python-setup-interpreter-constraints=CPython>=2.7,<3',
-               '--python-setup-interpreter-constraints=CPython>=3.3',
+               '--python-setup-interpreter-constraints=["CPython>=2.7,<3", ">=3.3"]',
```

Comment:
Did this fix the local test for you? Mine is still failing.

Reply:
No, it was an attempt do so, though.

Reply:
We can revert this change. This only fails locally and we can follow up separately. I suspect this is unrelated to this change and has to do with the crypto build.